feat(security): Documentation about the new requests Sanitizer. (#2565)
bonitasoft/bonita-doc: modules/security/pages/sanitizer-security.adoc

> = Requests Sanitizer

Reviewer: ```suggestion
= Input Validation (Requests Sanitizer)
```

:description: Bonita REST API is secured against XSS attacks with a requests sanitizer.

{description}

Cross-Site Scripting (also known as XSS), is a common example of web security vulnerability.
An attacker can compromise the interactions that users have with a vulnerable application.
By injecting some malicious code then executed on a user's browser, the attacker can impersonate this user and take illegitimate control on the application.

> To prevent such attacks, Bonita REST API is secured with a requests sanitizer which neturalizes malicious code from the body.

Reviewer: ```suggestion
To prevent such attacks, Bonita REST API is secured with a requests sanitizer which neutralizes malicious code from the body of incoming requests.
```

== Default security setup in Bonita

In Bonita, the UI screens are secured in order to prevent XSS attacks.

Since version 2022.1u8 (7.14.8) in 2024, in addition to UI screens protections, a Sanitizer protection is enabled by default to filter harmfull payloads out of API requests.
This additional security helps preventing stored XSS attacks on screens which may have not been secured enough, such as custom pages you have developed.

== How it works

On potentially harmfull REST API requests, a filter executes the https://github.com/OWASP/java-html-sanitizer/blob/master/README.md[OWASP Java HTML Sanitizer] on the request body.

> Malicious code is then removed or escaped from the request body before the execution.

Reviewer: ```suggestion
Malicious code is then removed or escaped from the request body before the request is handled.
```

When you migrate from a previous version, since the Sanitizer protection filters the request's inputs, it will not protect your unsecured UI screens from XSS attacks which have already been stored in the database.

== Is there a functional impact?

If you are displaying code in Bonita Applications, this may have a functional impact for you.
The code you wish to just display may be considered as some malicious injected code and removed from the input, or some characters may be replaced by their HTML equivalent, which may make it harder to edit afterward.

== How do I disable it?

> Disabling the Sanitizer protection is not recommended as it will make Bonita REST API unprotected from malicious attacks.

Reviewer: ```suggestion
Disabling the Sanitizer protection is not recommended as it will remove any server-side input validation and can make Applications using Bonita REST API unprotected from malicious attacks (unless content is sanitized front-end side).
```

But if you need to, the Sanitizer protection can be deactivated in the file `security-config.properties` for the whole platform.
The default version of this file is located in `setup/platform_conf/initial/platform_portal`. In order to change the configuration on an installation whose platform has already been initialized, use the xref:runtime:bonita-platform-setup.adoc[platform setup tool] to retrieve the current configuration and update the file in `setup/platform_conf/current/platform_portal`. Then use the tool again to save your changes into to the database.

When Sanitizer protection is enabled, this file contains this line:
`security.sanitizer.enabled true`

To disable the Sanitizer protection, edit the configuration file and change the `security.sanitizer.enabled` value to *false*.

> Make sure all your UI screens are secured against XSS attacks before disabling the Sanitizer protection, e.g. by developing them with the UI Designer.

Reviewer: ```suggestion
Make sure all your UI pages are secured against XSS attacks before disabling the Sanitizer protection (e.g. by developing them with Bonita UI Designer and avoiding using the `allow HTML` option of the widgets when the content comes from a user input).
```
